src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -117,21 +119,28 @@ build_mhd_response(const struct sol_http_response *response)
 
     SOL_HTTP_PARAM_FOREACH_IDX (&response->param, value, idx) {
         int ret;
-        char buffer[512];
+        char buffer[512], key[512], value_str[512];
```

Comment:
To be honest, I don't like the "fixed size buffer approach", I did not change it before, because I would like to make less changes as possible. However, the current approach can lead us to annoying bugs.
I will change the code and start using sol_buffer instead.